streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -70,6 +71,7 @@
 import static org.apache.kafka.streams.internals.StreamsConfigUtils.ProcessingMode.EXACTLY_ONCE_V2;
 import static org.apache.kafka.streams.processor.internals.StateManagerUtil.parseTaskDirectoryName;
 
+@SuppressWarnings("ClassFanOutComplexity")
```

Comment:
We are open for suggestion to solve this :)

Reply:
eh, I wouldn't worry too much about it...I mean yes, the TaskManager class is too complex, but that's not on you to solve. Tbh I'm surprised this is what pushed it over the edge

Reply:
Well actually after reading through the whole PR, I do think it's unnecessary complexity for the `TaskManager` after all -- see [this comment](https://github.com/apache/kafka/pull/14735#discussion_r1391855192). But that's more because of the complexity of determining when to invoke `#onUpdateSuspended` and to keep everything in the StoreChangelogReader class. Hopefully that all makes sense to you too